sub-packages/bionemo-esm2/src/bionemo/esm2/run/recipes.py

```diff
@@ -0,0 +1,447 @@
+        num_devices=8,
+    )
+
+
+def esm2_3b_model_config(initial_ckpt_path=None) -> ExposedESM2PretrainConfig:
+    """Model config for ESM2 3b."""
+    return ExposedESM2PretrainConfig(
+        num_layers=36,
+        hidden_size=2560,
+        ffn_hidden_size=2560 * 4,
+        num_attention_heads=40,
+        seq_length=1024,
+        biobert_spec_option=BiobertSpecOption.esm2_bert_layer_with_transformer_engine_spec,
+        initial_ckpt_path=initial_ckpt_path,
+        params_dtype="bf16-mixed",
+        pipeline_dtype="bf16-mixed",
+        autocast_dtype="bf16-mixed",
+    )
+
+
+def esm2_3b_wandb_config() -> WandbConfig:
+    """Wandb config for ESM2 3b."""
+    return WandbConfig(
+        entity="esm2-3b_pretraining",
+        project="esm2-3b_pretraining",
+        group="esm2-3b",
+        tags=["esm2-3b"],
+        offline=True,
+        anonymous=True,
+        id="1",
+        log_model=False,
+    )
+
+
+def esm2_3b_recipe(args) -> MainConfig[ExposedESM2PretrainConfig, ESM2DataConfig]:
+    """Recipe for ESM2 3b."""
+    data_config = ESM2DataConfig(
+        min_seq_length=1024,
+        max_seq_length=1024,
+        micro_batch_size=1,
+        num_dataset_workers=8,
+        train_cluster_path=args.train_cluster_path,
+        train_database_path=args.train_database_path,
+        valid_cluster_path=args.valid_cluster_path,
+        valid_database_path=args.valid_database_path,
+    )
+
+    return MainConfig(
+        data_config=data_config,
+        parallel_config=esm2_3b_parallel_config(),
+        training_config=esm2_base_training_config(),  # no changes for 8m
+        bionemo_model_config=esm2_3b_model_config(args.initial_ckpt_path),
+        optim_config=esm2_base_optimizer_scheduler_config(),  # no changes for 8m
+        experiment_config=esm2_650m_experiment_config(args.result_dir),
+        wandb_config=esm2_3b_wandb_config(),
+    )
+
+
+def simple_parallel_recipe(
+    tensor_model_parallel_size: int = 1,
+    pipeline_model_parallel_size: int = 1,
+    num_devices: int = 1,
+    accumulate_grad_batches: int = 1,
+) -> ParallelConfig:
+    """Simple parallel recipe for ESM2."""
+    assert (
+        num_devices >= tensor_model_parallel_size * pipeline_model_parallel_size
+    ), "devices must be divisible by tensor_model_parallel_size * pipeline_model_parallel_size"
+    return ParallelConfig(
+        tensor_model_parallel_size=tensor_model_parallel_size,
+        pipeline_model_parallel_size=pipeline_model_parallel_size,
+        num_devices=num_devices,
+        accumulate_grad_batches=accumulate_grad_batches,
+    )
+
+
+def tiny_train_config_recipe() -> TrainingConfig:
+    """Tiny training config for ESM2."""
+    return TrainingConfig(max_steps=10, limit_val_batches=2, val_check_interval=2)
+
+
+def default_adam_optimizer_with_cosine_annealing_recipe() -> OptimizerSchedulerConfig:
+    """Default optimizer scheduler config for ESM2."""
+    return OptimizerSchedulerConfig()
```

Comment:
ESM-2 doesn't use cosine annealing.

Reply:
OK- I didnt realize. I assumed we could get away with a naive approach for the initial PR, but seems like we will need something more complicated.

For now, I threw the kwargs for both cosine annealing and warmup into the same config. We should iterate on this asap after the release.